### PR TITLE
packer: use upstream vendoring

### LINKED ISSUE #2325: packer fails to brew
### Bug reports:

I am brewing `brew install packer` and getting 404 on a code.google.com repo. The url itself redirects to https://code.google.com/archive/p/gosshold/ as Google has shut down the service. There is no repository to clone at either of those urls and I was unable to find the correct url.

(I am using linuxbrew but that is presumably not the problem here.)

```
==> Cloning https://github.com/mitchellh/multistep.git
Cloning into '/var/lib/jenkins/.cache/Homebrew/packer--github.com-mitchellh-multistep--git'...
remote: Counting objects: 116, done.
remote: Total 116 (delta 0), reused 0 (delta 0), pack-reused 116
Receiving objects: 100% (116/116), 22.73 KiB | 0 bytes/s, done.
Resolving deltas: 100% (64/64), done.
Checking connectivity... done.
==> Checking out revision 162146fc57112954184d90266f4733e900ed05a5
==> Cloning https://code.google.com/p/gosshold/
abort: HTTP Error 404: Not Found
Error: Failed to download resource "packer--code.google.com-p-gosshold"
```

### PR DESCRIPTION
**Pull request by DomT4**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

CC @ilovezfs for a laugh at the diff size.

Closes #2325.

### CODE REVIEW

File: Formula/packer.rb [hunk 1]
```diff
@@ -22,340 +22,46 @@ class Packer < Formula
...
-    # get installed permanently
+    # For the gox buildtool used by packer, which
+    # doesn't need to be installed permanently.
     ENV.append_path "PATH", buildpath
 
     packerpath = buildpath/"src/github.com/mitchellh/packer"
     packerpath.install Dir["{*,.git}"]
     Language::Go.stage_deps resources, buildpath/"src"
-    mkdir_p buildpath/"bin"
+    (buildpath/"bin").mkpath
```
Comment (ilovezfs) [inline Formula/packer.rb]: Can we just `mkdir "bin"`

**Comment by ilovezfs** [comment]
@DomT4 Nice! And yet … my smile instantly vanishes when I see that it still has `go_resource` blocks hehe

File: Formula/packer.rb (hunk 1 above)
Comment (DomT4) [inline Formula/packer.rb]: I suspect so, but since we reference `buildpath` everywhere else in this initial block I bowed to consistency, heh.

**Comment by DomT4** [comment]
Yeah, the few remaining ones are a bit annoying. Progress, though. Hopefully even the ones remaining will disappear eventually.

**Pull request closed without merging**